### ISSUE TITLE
yolo26 onnx inference issue

### ISSUE THREAD

**Issue by hooNpk** [ultralytics/ultralytics#23955]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

Hello. I’m not sure whether this is a problem with yolosharp or with YOLO, anyway I’m posting it here.

When training a detection model with the yolo26 model,
I set a value for max_det (the default is 300) before training.
After exporting the model to ONNX and running inference with yolosharp,
the following error appears.
If I do not supply a separate max_det value (i.e., the default 300 is used), yolosharp works without any issues.

Could you investigate this issue? I suspect that the problem occurs because the layer changes when exporting the YOLO model to ONNX.

Error Log
Index was outside the bounds of the array. at Compunet.YoloSharp.Decoders.Base.AnchorBasedBoundingBoxDecoder.Decode(MemoryTensor1 tensor) in //src/YoloSharp/Decoders/Base/AnchorBasedBoundingBoxDecoder.cs:line 10 at Compunet.YoloSharp.Decoders.DetectionDecoder.Decode(IYoloRawOutput output, Size size) in //src/YoloSharp/Decoders/DetectionDecoder.cs:line 9 at Compunet.YoloSharp.YoloPredictor.Predict[T](Image1 image, YoloConfiguration configuration) in //src/YoloSharp/Predictor/YoloPredictor.cs:line 51 at Compunet.YoloSharp.YoloPredictorExtensions.Detect(YoloPredictor predictor, Image1 image, YoloConfiguration configuration) in //src/YoloSharp/Predictor/YoloPredictorExtensions.cs:line 87 at Compunet.YoloSharp.YoloPredictorAsyncExtensions.<>c__DisplayClass22_0.b__0() in /_/src/YoloSharp/Predictor/YoloPredictorAsyncExtensions.cs:line 92 at System.Threading.Tasks.Task1.InnerInvoke() at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state) --- End of stack trace from previous location --- at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state) at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread) --- End of stack trace from previous location --- at OnnxInferencer.Services.YoloObjectDetectionBb.InferAsync(Byte[] bgrImage, Int32 inputWidth, Int32 inputHeight) in C:\project\davinci_repo\Davinci\OnnxInferencer\Services\YoloObjectDetectionBb.cs:line 72

### Environment

OS : Windows 11 24H2
Python : 3.12.10
ultralytics : 8.4.21
onnxruntime-gpu : 1.23.2
torch : 2.5.1+cu124
torchvision : 0.20.1+cu124

### Minimal Reproducible Example

It's basic codes in official document.

### Training
#python
model.train(data=data_path, epochs=300, imgsz=image_size, device=0, max_det=1)

### Export
#python
model.export(format="onnx", imgsz=image_size, max_det=1)

### Inference
// C#
_predictor = new YoloPredictor(model, options);
var yoloResult = await _predictor.DetectAsync(rgbImage, yoloConfiguration);

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @hooNpk, thank you for reporting this possible YOLO26 ONNX inference issue 🚀 This is an automated response to help gather the right details, and an Ultralytics engineer will also assist you soon.

We recommend a quick visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many common questions may already be answered.

Since this appears to be a 🐛 Bug Report, please provide a complete [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate. You’ve already shared helpful snippets 🙌, but to reproduce this reliably please include:
- The exact training code used to create the YOLO26 model
- The exact export command/code used for ONNX
- Whether the same exported ONNX model runs correctly in Python before using YoloSharp
- The model summary or export log showing output shape details
- A minimal sample image and the exact YoloSharp configuration used at inference time
- Confirmation that the issue occurs specifically when `max_det` is set to a non-default value like `1`

If this is also related to custom training ❓, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Please upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify this is not already resolved in a newer release:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Looks like a YoloSharp issue because it's hardcoded to only work with `max_det=300`

**Comment by hooNpk** [comment]
Thanks for letting me know. I'll report this issue on yolosharp.

**Comment by glenn-jocher** [comment]
Sounds good — `max_det` is part of the exported end-to-end postprocess, so changing it can change the ONNX output shape, which matches a decoder-side assumption in YoloSharp; see [`Detect.postprocess()`](https://docs.ultralytics.com/reference/nn/modules/head/#ultralytics.nn.modules.head.Detect.postprocess). If you can reproduce the same behavior with native Ultralytics/Python ONNX inference, please share a small [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum-reproducible-example/) here along with `yolo checks` output and we’ll take another look.
